bridges/snowbridge/primitives/router/src/inbound/mod.rs

```diff
@@ -289,6 +317,53 @@ where
 			[GlobalConsensus(network), AccountKey20 { network: None, key: token.into() }],
 		)
 	}
+
+	fn convert_transact(
+		chain_id: u64,
+		sender: H160,
+		origin_kind: OriginKind,
+		fee: u128,
+		weight_ref_time: u64,
+		weight_proof_size: u64,
+		payload: Vec<u8>,
+	) -> (Xcm<()>, Balance) {
+		let xcm_fee: Asset = (Location::parent(), fee).into();
+
+		let xcm: Xcm<()> = vec![
+			// Only our inbound-queue pallet is allowed to invoke `UniversalOrigin`
+			DescendOrigin(PalletInstance(InboundQueuePalletInstance::get()).into()),
+			// Change origin to the bridge.
+			UniversalOrigin(GlobalConsensus(Ethereum { chain_id })),
+			// DescendOrigin to the sender.
+			DescendOrigin(AccountKey20 { network: None, key: sender.into() }.into()),
+			// Withdraw fees from sender to pay the xcm execution
+			WithdrawAsset(xcm_fee.clone().into()),
+			// Pay for execution.
+			BuyExecution { fees: xcm_fee.clone(), weight_limit: Unlimited },
+			// Transact on dest chain.
+			Transact {
+				origin_kind,
+				require_weight_at_most: Weight::from_parts(weight_ref_time, weight_proof_size),
+				call: payload.into(),
+			},
+			ExpectTransactStatus(MaybeErrorCode::Success),
```

Comment:
I'm not sure this instruction is useful. Typically this would be used to halt execution of the message.

The only thing its doing right now is preventing subsequent refund operations from executing.

Reply:
https://github.com/Snowfork/polkadot-sdk/pull/114/commits/9968a37a69ba1b2e1439aacf9025d962b9849e8a